File: BuilderDesignPattern/CarManufacture_Example/Components/TripComputer.java
package BuilderDesignPattern.CarManufacture_Example.Components;

import BuilderDesignPattern.CarManufacture_Example.Products.Car;

public class TripComputer {
	private Car car;

	public void setCar(Car car) {
		this.car = car;
	}

	public void showFuelLevel() {
		System.out.println("Fuel Level: " + car.getFuel());
	}

	public void showStatus() {
		if (this.car.getEngine().isStarted()) {
			System.out.println("Car is Started.");
		} else {
			System.out.println("Car is not Started.");
		}
	}
}

File: BuilderDesignPattern/CarManufacture_Example_Python/Components/TripComputer.py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from BuilderDesignPattern.CarManufacture_Example_Python.Products.Car import Car


class TripComputer:
    def __init__(self):
        self._car: "Car" = None

    def setCar(self, car: "Car"):
        self._car = car

    def showFuelLevel(self):
        if self._car:
            print(f"Fuel Level: {self._car.getFuel()}")
        else:
            print("Trip Computer not connected to a car.")

    def showStatus(self):
        if self._car.getEngine().isStarted():
            print(f"Car is Started.")
        else:
            print(f"Car is Turned off.")